core/src/main/java/org/elasticsearch/env/Environment.java

```diff
@@ -332,4 +339,26 @@ public Path tmpFile() {
     public static FileStore getFileStore(Path path) throws IOException {
         return ESFileStore.getMatchingFileStore(path, fileStores);
     }
+
+    /**
+     * assserts that the two environments are equivalent for all things the environment cares about (i.e., all but the setting
+     * object which may contain different setting)
+     */
+    public static void assertEquivalent(Environment actual, Environment expected) {
```

Comment:
Okay. If you have a plan to remove settings from `Environment`, then I think we can do this? Leave the method here. When you remove settings, change the method to be the implementation of `equals` (and add an implementation of `hashCode`). What do you think?